### PR TITLE
Group Alpha rec sys app

### PR DESCRIPTION
**Pull request by angihe93**
This is our finished app

### CODE REVIEW

File: services/backend/src/data_structures/NLP_Tagger_Alpha/__init__.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    global OFFENSIVE_BLACKLIST 
+    if os.path.isfile("df_prompt_labeled.pkl"):
+        print("reading NLP label data")
+        with open("df_prompt_labeled.pkl", "rb") as f:
+            data = pickle.load(f)
+            OFFENSIVE_BLACKLIST = list(data.loc[data.offensive == True].index)
+    else:
+        print("The NLP Label data is missing")
+
+    return 0
```
Comment (kennethgoodman) [inline services/backend/src/data_structures/NLP_Tagger_Alpha/__init__.py:25]: no need for a return

File: services/backend/src/recommendation_system/ml_models/trained_model/training.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    dataset = dataset.drop('label',axis=1)
+    dt = tfdf.keras.pd_dataframe_to_tf_dataset(dataset, task=tfdf.keras.Task.REGRESSION)
+    predicted = model.predict(dt)
+    print(predicted)
+    print(predicted.max())
+    print(sum(predicted))
+    print(len(predicted))
+
+
+# if __name__ == "__main__":
```
Comment (kennethgoodman) [inline services/backend/src/recommendation_system/ml_models/trained_model/training.py:147]: delete commented code

File: services/backend/src/recommendation_system/recommendation_flow/controllers/AlphaController.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# from src.recommendation_system.recommendation_flow.candidate_generators.RandomGenerator import (
```
Comment (kennethgoodman) [inline services/backend/src/recommendation_system/recommendation_flow/controllers/AlphaController.py:1]: delete commented code

File: services/backend/src/recommendation_system/recommendation_flow/controllers/AlphaController.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+)
+
+
+class AlphaController(AbstractController):
+    def get_content_ids(self, user_id, limit, offset, seed, starting_point):
+        print('limit from AlphaController get_content_ids():',limit)
+        if seed <= 1:  # MySql seeds should be [0, # of rows] not [0, 1]
+            seed *= 1000000
+        candidates_limit = (
+            limit * 10 * 10
```
Comment (kennethgoodman) [inline services/backend/src/recommendation_system/recommendation_flow/controllers/AlphaController.py:33]: does this make sense in your context?

File: services/backend/src/recommendation_system/recommendation_flow/model_prediction/SitongModel.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
+import random
```
Comment (kennethgoodman) [inline services/backend/src/recommendation_system/recommendation_flow/model_prediction/SitongModel.py:1]: if this is someone's model, please still label it with Alpha first. Ideally in a folder named Alpha

Comment (kennethgoodman) [reply services/backend/src/recommendation_system/recommendation_flow/model_prediction/SitongModel.py:1]: also is this file used anywhere?

File: services/backend/src/recommendation_system/recommendation_flow/model_prediction/AlphaModel.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            subprocess.check_call([sys.executable, "-m", "pip", "install", "tensorflow_decision_forests"])
+            import tensorflow_decision_forests as tfdf
+
+            with open('/usr/src/app/gbdt_model_v3.pickle', 'rb') as f:
+                GBDT_model = pickle.load(f)
+            with open('/usr/src/app/content_artist_style_dic.pickle', 'rb') as f:
+                dic_id_style = pickle.load(f)
+            with open('/usr/src/app/prediction_prep_dic.pickle', 'rb') as f:
+                prep_dic = pickle.load(f)
+            with open('/usr/src/app/dic_id_to_embedding.pickle', 'rb') as f:
```
Comment (kennethgoodman) [inline services/backend/src/recommendation_system/recommendation_flow/model_prediction/AlphaModel.py:29]: nit: https://stackoverflow.com/questions/893333/multiple-variables-in-a-with-statement

File: services/backend/src/recommendation_system/recommendation_flow/model_prediction/AlphaModel.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+import pickle
+import sys
+import subprocess
+from .AbstractModel import AbstractModel
+
+
+class AlphaModel(AbstractModel):
+    def predict_probabilities(self, content_ids, user_id, seed=None, **kwargs):
+        
+        try:
```
Comment (kennethgoodman) [inline services/backend/src/recommendation_system/recommendation_flow/model_prediction/AlphaModel.py:17]: nit: this try-except block is too large, which line do you think will have the exception, only put those lines in the try-catch or refactor into a function call

File: services/backend/src/recommendation_system/recommendation_flow/ranking/AlphaRanker.py [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+                    others.append(i)
+            top_other = heapq.nlargest(other, others, key=lambda x: x["explore"])
+            # if len(top_other) < other:
+            #     handle = top_other + top_limit[k:limit]
+            #     handle2 = sorted(set(handle), key=handle.index)
+            #     top_other = handle2[:other]
+
+            top_all = top_k + top_other
+
+        except KeyError:
```
Comment (kennethgoodman) [inline services/backend/src/recommendation_system/recommendation_flow/ranking/AlphaRanker.py:28]: when does a keyerror happen?

**Review by kennethgoodman — COMMENTED** [review]
1. delete commented code
2. don't delete files
3. dont put "2" on files
4. if a file is specific to your group, please put it in a folder with "Alpha" in it or within the file name
5. if a file isn't used anywhere in production (only for training/analysis), lets put it in another folder, say services/backend/offline/alpha

Good job, looking forward to testing it out, please make the changes then I will approve and deploy

**Comment by angihe93** [comment]
Thank you for the feedback. I've made the changes and made a [new commit](https://github.com/angihe93/Columbia-E4579-alpha/commit/59e4ec16a2d630aa68d69671384d1911ad0d8d37) to our forked repo. What we did specifically regarding your comments: 
1. delete commented code: done
2. don't delete files: i don't remember deleting any files that came with the app, but could have done so accidentally. please let me know if anything is missing and i'll add it back in. 
3. dont put "2" on files: i think some "2" files came with the fork. i've deleted those since they are duplicates.  
4. if a file is specific to your group, please put it in a folder with "Alpha" in it or within the file name: our generator, filter, prediction model, ranker, controller all start with Alpha in the file name. we've moved our other files to services/backend/src/alpha.
5. if a file isn't used anywhere in production (only for training/analysis), lets put it in another folder, say services/backend/offline/alpha: done

**Comment by kennethgoodman** [comment]
Please see "files changed", it says 82 with 3037 lines added and 14,533 deleted for deleted/added code

**Comment by kennethgoodman** [comment]
You will also need to resolve conflicts with the 3 files above

**Comment by angihe93** [comment]
OK I think the deleted files might have been from a git lfs/git ignore issue. I've added back the files in services/backend/seed_data/data, and resolved the conflicts.

**Comment by kennethgoodman** [comment]
Please fix all the "2" at the end of files and creating new files that are unnecessary, you can see this in "Files changes"

**Comment by kennethgoodman** [comment]
Also there are still commented code, also please remove print statements that you added.

It may be easiest/simplest to:
1. re-fork the repo
2. clone the repo
3. add the specific changes you need, and only those changes
4. commit
5. push
6. create pull request

See https://github.com/kennethgoodman/Columbia-E4579/pull/51/commits/2a52551d67303b81ee213725e58d5562abe56723 for the commit that looks like it added these changes

**Comment by ChuyangXiao** [comment]
Thank you for your suggestions. Nevertheless, since we have deleted the repeated files, we are both wondering what files are you talking about when you suggest the "unnecessary" files with 2 at their ends. Thanks

**Comment by kennethgoodman** [comment]
Please see “Files changed”, there are 81, do you not see the additional files?For example, services/backend/src/api/content/views 2.pySent from my iPhoneOn Dec 19, 2022, at 4:57 PM, Chuyang Xiao ***@***.***> wrote:﻿
Thank you for your suggestions. Nevertheless, since we have deleted the repeated files, we are both wondering what files are you talking about when you suggest the "unnecessary" files with 2 at their ends. Thanks
